### PR TITLE
Comfig composer and routes admin

### PR DESCRIPTION
**Pull request by at-huydoan**
add package and route for backend and frontend

### CODE REVIEW

File: app/Http/Routes/Dashboard.php [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<?php
+    Route::get('/', 'DashboardController@index');
```
Comment (ghost) [inline app/Http/Routes/Dashboard.php]: How many routes will be in this file ?

**Comment by ghost** [comment]
Why did you change file permission from `644` to `755` ?

File: app/Providers/RouteServiceProvider.php [hunk 2]
```diff
@@ -33,13 +39,19 @@ public function boot(Router $router)
...
+     * @param Router  $router  router
+     * @param Request $request router
      *
      * @return void
      */
-    public function map(Router $router)
+    public function map(Router $router, Request $request)
     {
-        $router->group(['namespace' => $this->namespace], function ($router) {
+        if ($request->segment(1) == self::ADMIN_PATH) {
```
Comment (ghost) [inline app/Providers/RouteServiceProvider.php]: it's not good to check `admin` in URL and add namespace. You have to add namespaces for both `Admin` and `Frontend`

File: app/Providers/RouteServiceProvider.php [hunk 3]
```diff
@@ -33,13 +38,17 @@ public function boot(Router $router)
...
      */
-    public function map(Router $router)
+    public function map(Router $router, Request $request)
     {
-        $router->group(['namespace' => $this->namespace], function ($router) {
+        $router->group(['namespace' => $this->namespace[0]], function ($router) {
+            require app_path('Http/Backend/routes.php');
+        });
+        $router->group(['namespace' => $this->namespace[1]], function ($router) {
             require app_path('Http/routes.php');
```
Comment (ghost) [inline app/Providers/RouteServiceProvider.php]: remove `Http/routes.php` add to `Http/Frontend/routes.php`

File: app/Providers/RouteServiceProvider.php [hunk 4]
```diff
@@ -33,13 +38,17 @@ public function boot(Router $router)
...
+     * @param Router  $router  router
+     * @param Request $request router
      *
      * @return void
      */
-    public function map(Router $router)
+    public function map(Router $router, Request $request)
     {
-        $router->group(['namespace' => $this->namespace], function ($router) {
+        $router->group(['namespace' => $this->namespace[0]], function ($router) {
```
Comment (ghost) [inline app/Providers/RouteServiceProvider.php]: `$this->namespace[0]` is not good. Creating a property is better

File: app/Providers/RouteServiceProvider.php [hunk 5]
```diff
@@ -14,7 +15,11 @@ class RouteServiceProvider extends ServiceProvider
      *
      * @var string
      */
-    protected $namespace = 'App\Http\Controllers';
+    protected $namespace = [
```
Comment (ghost) [inline app/Providers/RouteServiceProvider.php]: just create 2 properties: `$feNamespace` and `$beNamespace`

File: app/Providers/RouteServiceProvider.php [hunk 6]
```diff
@@ -33,13 +38,17 @@ public function boot(Router $router)
...
      * @return void
      */
-    public function map(Router $router)
+    public function map(Router $router, Request $request)
     {
-        $router->group(['namespace' => $this->namespace], function ($router) {
+        $router->group(['namespace' => $this->namespace[0]], function ($router) {
+            require app_path('Http/Backend/routes.php');
+        });
+        $router->group(['namespace' => $this->namespace[1]], function ($router) {
```
Comment (ghost) [inline app/Providers/RouteServiceProvider.php]: add route prefix into this group is better

**Comment by ghost** [comment]
@thienvuong Please check my comment

File: app/Providers/RouteServiceProvider.php [hunk 7]
```diff
@@ -33,13 +35,17 @@ public function boot(Router $router)
...
      */
-    public function map(Router $router)
+    public function map(Router $router, Request $request)
     {
-        $router->group(['namespace' => $this->namespace], function ($router) {
+        $router->group(['prefix' => 'admin', 'namespace' => $this->beNamespace], function ($router) {
+            require app_path('Http/Backend/routes.php');
+        });
+        $router->group(['namespace' => $this->feNamespace], function ($router) {
             require app_path('Http/routes.php');
```
Comment (ghost) [inline app/Providers/RouteServiceProvider.php]: why don't you create a route file in `Http/Frontend/routes.php` ?

**Comment by at-huydoan** [comment]
please check, this required for next commit

**Comment by ghost** [comment]
LGTM, thanks

**Pull request merged**